### DISCUSSION TITLE
Should `ki` force the return value of a thread to WHNF?

### DISCUSSION THREAD

**Discussion by mitchellwrosen** [awkward-squad/ki#32] [General]
I wonder if `ki` should force a thread's result to WHNF before returning.

This would be inconsistent with `async`, and probably other threading libraries, so by the principle of least surprise, maybe we shouldn't do this.

However, it might be nice to assist users in performing (some) computation on the dedicated thread they explicitly spawn, rather than let them accidentally conclude a background thread with some expensive thunk that's computed by the first awaiter who peeks at it.

**Comment by ocharles** [comment]
One option might be to add `fork'`, using the convention that `'` means strict
